src/anyio/_core/_tasks.py

```diff
@@ -565,7 +565,7 @@ async def as_completed(
         yield iterator
 
 
-async def gather(*coros: Coroutine[Any, Any, Any]) -> Sequence[Any]:
+async def gather(*coros: Coroutine[Any, Any, T]) -> Sequence[T]:
```

Comment:
Eventually overrides will be necessary here to allow mixed return types. Also looks like it returns `list[T]` not `Sequence`

Reply:
Yes, the point is to be more specific since this returns a list. However, now that I think about it, this should return a tuple.

Reply:
https://github.com/python/typeshed/blob/main/stdlib/asyncio/tasks.pyi#L229

Reply:
`asyncio.gather()` also always returns a list. The annotations are therefore deceptive.

Reply:
Also frankly what's the point of converting the result to a tuple? There's no requirement for it to be read-only. In fact if the caller then needs to mangle the result somehow they'll need to convert it right back.

Reply:
I would rather the types be consistent, but that's just my preference I guess

Reply:
Copying the types from `typeshed` would make them inconsistent.